.github/workflows/ci-test-integration.yml

```
@@ -37,6 +37,23 @@ jobs:
+      - name: Build test-runner binary
+        run: cargo build --package test-runner --bin run-tests --locked
+        shell: bash
+        working-directory: magicblock-validator
+
+      - name: Pre-build all test binaries
+        run: cargo build --tests --locked
+        shell: bash
+        working-directory: magicblock-validator/test-integration
+
+      - name: Upload test-runner binary
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.9)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)